tests/cpptests/test_cpp_thpool.cpp

```diff
@@ -207,6 +304,293 @@ TEST_P(PriorityThpoolTestFunctionality, TestPauseResume) {
+    redisearch_thpool_wait(this->pool);
+    // Expect 0 threads alive and n_jobs done.
+    stats = redisearch_thpool_get_stats(this->pool);
+    while (stats.num_threads_alive) {
+        usleep(1);
+        stats = redisearch_thpool_get_stats(this->pool);
+    }
+    ASSERT_EQ(stats.total_jobs_done, total_jobs_pushed) << "expected " << total_jobs_pushed << " jobs done";
+    ASSERT_EQ(stats.total_pending_jobs, 0) << "expected " << 0 << " jobs pending";
+}
+
+/** This test purpose is to show we can add threads to a pool that was set to 0 threads. */
+TEST_P(PriorityThpoolTestRuntimeConfig, TestAddThreadsToEmptyPool) {
+    size_t total_jobs_pushed = 0;
+    // Add a job to trigger thpool initialization
+    size_t time_us = 1;
+    redisearch_thpool_add_work(this->pool, sleep_job_us, &time_us, THPOOL_PRIORITY_HIGH);
+    ++total_jobs_pushed;
+    ASSERT_EQ(redisearch_thpool_get_stats(this->pool).num_threads_alive, RUNTIME_CONFIG_N_THREADS) << "expected " << RUNTIME_CONFIG_N_THREADS << " threads alive";
+    ASSERT_TRUE(redisearch_thpool_is_initialized(this->pool)) << "expected thread pool to be initialized";
+    redisearch_thpool_wait(this->pool);
+
+    // Remove all threads.
+    ASSERT_EQ(redisearch_thpool_remove_threads(this->pool, RUNTIME_CONFIG_N_THREADS), 0) << "expected 0 n_threads";
+    // Threadpool is still initialized, but no threads are alive.
+    ASSERT_TRUE(redisearch_thpool_is_initialized(this->pool)) << "expected thread pool to be initialized";
+    ASSERT_EQ(redisearch_thpool_get_stats(this->pool).num_threads_alive, 0) << "expected 0 threads alive";
+
+    // Add threads.
+    ASSERT_EQ(redisearch_thpool_add_threads(this->pool, RUNTIME_CONFIG_N_THREADS), RUNTIME_CONFIG_N_THREADS) << "expected " << RUNTIME_CONFIG_N_THREADS << " n_threads";
+    ASSERT_EQ(redisearch_thpool_get_stats(this->pool).num_threads_alive, RUNTIME_CONFIG_N_THREADS) << "expected " << RUNTIME_CONFIG_N_THREADS << " threads alive";
+    ASSERT_TRUE(redisearch_thpool_is_initialized(this->pool)) << "expected thread pool to be initialized";
+
+    // Validate the thpool functionality.
+    redisearch_thpool_add_work(this->pool, sleep_job_us, &time_us, THPOOL_PRIORITY_HIGH);
+    ++total_jobs_pushed;
+    redisearch_thpool_wait(this->pool);
+    ASSERT_EQ(redisearch_thpool_get_stats(this->pool).total_jobs_done, total_jobs_pushed) << "expected " << total_jobs_pushed << " jobs done";
+}
+
+TEST_P(PriorityThpoolTestRuntimeConfig, TestWaitTerminate) {
+    size_t num_jobs = 100;
+    size_t time_ms = 100;
+    // add 100 jobs
+    redisearch_thpool_work_t jobs[num_jobs];
+    for (size_t i = 0; i < num_jobs; i++) {
+        jobs[i].arg_p = &time_ms;
+        jobs[i].function_p = sleep_job_ms;
+    }
+    redisearch_thpool_add_n_work(this->pool, jobs, num_jobs, THPOOL_PRIORITY_LOW);
+
+    // Wait for them to finish.
+    redisearch_thpool_wait(this->pool);
+    // Terminate threads.
+    redisearch_thpool_terminate_threads(this->pool);
+}
```

Comment:
It simulates burst threads scenario. I used it to try reproduce the bug. 
Since we have other tests to cover terminate when empty scenarios, and burst threads flow is covered in flow tests, i can remove it now.